src/Compilers/CSharp/Portable/Parser/SyntaxParser.cs

```diff
@@ -1079,11 +1079,12 @@ protected bool IsFeatureEnabled(MessageID feature)
         ///     while (IsMakingProgress(ref tokenProgress))
         /// It should be used as a guardrail, not as a crutch, so it asserts if no progress was made.
         /// </summary>
-        protected bool IsMakingProgress(ref int lastTokenOffset)
+        protected bool IsMakingProgress(ref int lastTokenPosition)
         {
-            if (_tokenOffset > lastTokenOffset)
+            var currentPosition = _firstToken + _tokenOffset;
```

Comment:
>var currentPosition = _firstToken + _tokenOffset; [](start = 12, length = 49)

Would it make sense to add a property for this calculation and use it here and other similar places?